src/wp-includes/functions.php

```diff
@@ -9193,10 +9193,7 @@ function wp_unique_id_from_values( array $data, string $prefix = '' ): string {
 	if ( empty( $data ) ) {
 		_doing_it_wrong(
 			__FUNCTION__,
-			sprintf(
-				__( 'The $data argument must not be empty.' ),
-				gettype( $data )
-			),
+			__( 'The $data argument must not be empty.' ),
```

Comment:
@peterwilsoncc we should probably add a translators note to this as well before merging.

Reply:
@joemcgill What do you think of removing the DIW and allowing mixed types for the `$data`. I think that will make it more useful for third party developers wishing to generate unique IDs from various input types.

Reply:
The only way I see that working is if we pass off an empty `$data` value to either `wp_unique_prefixed_id()` or `wp_unique_id()` if there is no prefix. However, I think it's reasonable to expect a non-empty array in this function or to allow the hash to be generated as is with an empty array. Given that this function has strict types enforced, passing any data other than an array will already throw an error.

Reply:
Translater note added in https://github.com/WordPress/wordpress-develop/pull/8570/commits/c2c1f4eed97a965c09084961086494794197acfd